server/src/main/java/org/elasticsearch/persistent/UnassignPersistentTaskAction.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.persistent;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.MasterNodeOperationRequestBuilder;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.client.ElasticsearchClient;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class UnassignPersistentTaskAction extends ActionType<PersistentTaskResponse> {
+
+    public static final UnassignPersistentTaskAction INSTANCE = new UnassignPersistentTaskAction();
+    public static final String NAME = "cluster:internal/persistent/unassign";
```

Comment:
I have tried to write the code for this PR in the style of the pre-existing persistent tasks code.

I wanted to call out this line as it's the one intentional deviation from the pre-existing style.

Starting internal action names with `cluster:internal` instead of `cluster:admin` means that only `superuser` can call them via the transport client, so restricts access slightly more than just not having a REST endpoint.  It's something we've done to a few ML endpoints instead of using `internal:admin` because it means the endpoint can still be called by the internal `_xpack` user.  However, I realise that some of the other persistent tasks actions also fall into the "internal only" category and have action names beginning `cluster:admin`, so if you'd prefer me to make this one consistent with those then just let me know.